docs/migration-guides/v2/lib.md

```diff
@@ -0,0 +1,375 @@
+# Migration guide for ably-js v2
+
+Here’s how to migrate from ably-js v1 to v2:
+
+1. [Stop using functionality that v1 deprecated and which v2 removes](#stop-using-v1-deprecated).
+1. [Stop using other functionality that v2 removes](#stop-using-v2-removed).
+1. [Update to v2 and handle its breaking changes](#update-to-v2-and-handle-breaking-changes).
+1. (Optional) [Stop using functionality that v2 deprecates](#stop-using-v2-deprecated).
+1. (Optional) [Take advantage of new features that v2 introduces](#use-v2-new-features).
+
+<h2 id="stop-using-v1-deprecated">Stop using functionality that v1 deprecated and which v2 removes</h2>
+
+Begin by updating to ably-js 1.2.50 or later, to make sure you see the deprecation log messages [described below](#stop-using-v1-deprecated-all-users).
+
+Now, you need to stop using the functionality that is deprecated by v1 and which is removed in v2. Here we explain how.
+
+The changes below are split into those that affect all users, and those that only affect TypeScript users.
+
+<h3 id="stop-using-v1-deprecated-all-users">Changes that affect all users</h3>
+
+In ably-js 1.2.50 or later, use of the following APIs will trigger a deprecation warning at runtime.
+
+**Note about deprecation warnings:** These deprecation warnings take the form of error-level log messages emitted through the library’s logging mechanism (i.e. `ClientOptions.log.handler` or `ClientOptions.logHandler` if you’ve set these properties, or else `console.log()`). To find them in your logs, search for `Ably: Deprecation warning`.
+
+- Changes to `Crypto.getDefaultParams()`:
+  - The ability to pass a callback to this method has been removed in v2. This method now directly returns its result, instead of returning it asynchronously. Update your code so that it uses the return value of this method instead of passing a callback.
+  - The ability to call this method without specifying an encryption key has been removed in v2. Update your code so that it instead passes an object whose `key` property contains an encryption key. That is, replace `Crypto.getDefaultParams()` with `Crypto.getDefaultParams({ key })`, where `key` is an encryption key that you have generated (for example from the `Crypto.generateRandomKey()` method).
+  - The ability to pass the encryption key as the first argument of this method has been removed in v2. Update your code so that it instead passes an object whose `key` property contains the key. That is, replace `Crypto.getDefaultParams(key)` with `Crypto.getDefaultParams({ key })`.
+- The `headers` client option has been removed in v2. Remove your use of this client option.
```

Comment:
I think we could maybe order these bullets a bit better in order of how likely users are to be using them? for example, `log`/`logLevel` is quite a common and important option whereas I'd be quite surprised if anyone was using `headers`

Reply:
I agree that the logging one is likely to affect a lot of people, so have put that first. And since nobody internally even seemed to know about `headers`, I've put that last. Beyond that I have no idea what things are more likely than others so I’ve kept the order as it is :)